Shopify/cli
# Remove APIClient as it looks like it's not in use

### WHY are these changes introduced?

Remove APIClient as it looks like it's not in use. Reasons I think is not in use (beyond instantiation):

1. In Shopicy/cli/packages/ui-extensions-server-kit ExtensionServerClient is the only code to instantiate APIClient.  BUT! it does nothing with the instantiated code (e.g: use one of APIClient 's methods).
2. In Shopicy/cli/packages/ui-extensions-dev-console nothing uses the APIClient directly.
3. In Shopify/web nothing seemed to use the APIClient directly.

### WHAT is this pull request doing?

Removes all references to APIClient. As part of removing APIClient We get to:

1. Move filterExtensionsBySurface directly into ExtensionsServerClient, which is less overhead than a utilities folder and an index file (which suggests it's reused when it's not)
2. Simplify the typing of filterExtensionsBySurface.  Previously this had to satisfy two slightly different types.  Now it only has to simplify one.

### How to test your changes?

1. `yarn && yarn build`
3. `cd fixtures/app`
4. `yarn generate extension` (Choose the new UI Extension)
5. `yarn dev`
6. Open the dev console URL that the CLI outputs, e.g:

<img width="480" alt="Screenshot 2022-11-30 at 2 09 32 PM" src="https://user-images.githubusercontent.com/690791/204887053-f83baa6e-54a8-477f-ae43-20f0b4036f06.png">

7. Add `?embedded=true&surface=admin` to the URL.  You should see 2 extensions.
8. Remove the URL params.  You should see all extensions.
9. Click a preview link.  It should take you to 

Then let's perform some manual checks:

1. Scan `Shopify/web` for use of `ui-extensions-serve-kit`.  Try to find an instance of `extensions()` or `extensionById()` from `APIClient` being used. 
2. Scan `Shopify/cli/packages/ui-extensions-dev-console`.  Try to find an instance of `extensions()` or `extensionById()` from `APIClient` being used. 

### Post-release steps
None

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
